src/color.rs

```diff
@@ -0,0 +1,522 @@
+use std::fmt::Display;
+
+use auto_ops::impl_op_ex;
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[repr(C)]
+pub struct Color {
+    /// Red component [0.0, 1.0]
```

Comment:
I think we should support HDR colors, not just 0.0-1.0

Reply:
Maybe we should allow all ranges, to keep it simple.